app/src/main/java/com/example/duccm/learningapp/EditProfileDialogFragment.java

```diff
@@ -0,0 +1,216 @@
+        // Required empty public constructor
+    }
+
+    public interface OnSubmitButtonClickedListener {
+        public void passProfileData(Profile profile);
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+
+        try {
+            mCallback = (OnSubmitButtonClickedListener) context;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(context.toString()
+                    + " must implement OnSubmitButtonClickedListener");
+        }
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Light_NoTitleBar_Fullscreen);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_edit_profile_dialog, container, false);
+        getActivity().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
+        initViews(v);
+        setViews();
+        initListeners();
+        return v;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        Dialog dialog = super.onCreateDialog(savedInstanceState);
+        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        return dialog;
+    }
+
+    public void initViews(View v) {
+        avatar = (ImageView) v.findViewById(R.id.avatar);
+        editAvatar = (TextView) v.findViewById(R.id.edit_avatar);
+        name = (EditText) v.findViewById(R.id.edit_name);
+        email = (EditText) v.findViewById(R.id.edit_email);
+        phone = (EditText) v.findViewById(R.id.edit_phone);
+        birthday = (EditText) v.findViewById(R.id.edit_birthday);
+        update = (Button) v.findViewById(R.id.button_update);
+        cancel = (TextView) v.findViewById(R.id.text_cancel);
+    }
+
+    public void setViews() {
+        currentUser = new ProfileList().getProfileList().get(0);
+        Glide.with(this).load(currentUser.getAvatar()).into(avatar);
+        name.setText(currentUser.getName());
+        email.setText(currentUser.getEmail());
+        phone.setText(currentUser.getPhone());
+        birthday.setText(currentUser.getBirthday());
+    }
+
+    public void initListeners() {
+        editAvatar.setOnClickListener(this);
+        update.setOnClickListener(this);
+        cancel.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View view) {
+        switch (view.getId()) {
+            case R.id.edit_avatar:
+                checkPermissionAndTakePhoto();
+                break;
+            case R.id.button_update:
+                if (currentAvatarPath != null) {
+                    currentUser.setAvatar(currentAvatarPath);
+                }
+                currentUser.setName(name.getText().toString());
```

Comment:
mấy trường get từ edit text nhớ trim() lại nếu k đánh toàn dấu cách cũng pass đc đó